### PR TITLE
Fix to send X-GitHub-Event to SQS

### LINKED ISSUE #1121: Amazon SQS doesn't send event in message_attributes
# Problem
When using the SQS service, it doesn't send which event was fired as part of the Message Attributes, so it's hard to determine which service event was triggered.

# Solution
Change
```ruby
queue.send_message(clean_for_json(payload))
```
to this
```ruby
    sqs.send_message(
      queue_url: queue.url,
      message_body: clean_for_json(payload),
      message_attributes: {
        'X-Github-Event' => { string_value: event, data_type: 'String'}
      }  
    )
```

I can submit a PR, but i'm not sure if this is considered a bug fix or an improvement.

### PR DESCRIPTION
**Pull request by jwhitcraft**
Send the X-GitHub-Event to SQS as a message attribute so that users of SQS can know when event is in the Queue

Fixes #1121

### CODE REVIEW

**Comment by bswinnerton** [comment]
Hi @jwhitcraft, would it be possible to add a test that asserts the appropriate value is present for `X-GitHub-Event`?

File: lib/services/sqs_queue.rb [hunk 1]
```diff
@@ -41,7 +41,13 @@ def notify_sqs(aws_access_key, aws_secret_key, payload)
     else
         queue = sqs.queues.named(queue_name)
     end
-    queue.send_message(clean_for_json(payload))
+    sqs.send_message(
+      queue_url: queue.url,
+      message_body: clean_for_json(payload),
+      message_attributes: {
```
Comment (bswinnerton) [inline lib/services/sqs_queue.rb]: It's very possible that I'm wrong, but I'm not sure that [`message_attributes`](https://github.com/aws/aws-sdk-ruby/blob/1.27.0/lib/aws/sqs/queue.rb#L87-L115) is supported in [version 1.27.0](https://github.com/github/github-services/blob/c7ebee32771af3faed36500dfe239b1605478524/github-services.gemspec#L61) of aws-sdk-ruby.

Comment (jwhitcraft) [inline lib/services/sqs_queue.rb]: @bswinnerton, I just checked and i was using 1.66 locally which supports it but, 1.27.0 doesn't.  I've updated the code to use the aws-sdk-core (same as sns) and now it works correctly.

**Comment by jwhitcraft** [comment]
@bswinnerton I'm not sure how i can create a test for that part of the code without restructuring most of the class which I think is out of scope for this ticket.

**Comment by bswinnerton** [comment]
Hi @jwhitcraft,

> I'm not sure how i can create a test for that part of the code without restructuring most of the class which I think is out of scope for this ticket.

Unfortunately without any accompanying tests, we don't have a definitive way to verify that this pull request won't break the SQS integration for others. 

Given that this PR moves to the `aws-sdk-core` library, we'd really like to gain some level of confidence that the message is still sent, and with the new `X-GitHub-Event` attribute. That being said, I do think that accompanying tests would be in the scope of this ticket. Perhaps `aws-sdk-core` offers some [testing stubs](https://ruby.awsblog.com/blog/tag/Testing) that can be used.

**Comment by jwhitcraft** [comment]
@bswinnerton, i'll take a look at it this weekend to see if i can work something out.

**Comment by jwhitcraft** [comment]
@bswinnerton, I've added a test, but the one that is failing is also failing on master, i've filed a new issue for it

**Comment by bswinnerton** [comment]
Much thanks, @jwhitcraft. We'll investigate the failing test in `master`. I'm conferring with my colleagues regarding your latest changes. I'll follow up as soon as I can.

**Comment by bswinnerton** [comment]
Hi @jwhitcraft,

After conferring with the rest of the team responsible for github-services, we think it may be may be preferable to bump the `aws-sdk` gem to the earliest version that supports `message_attributes` rather than switching over to `aws-sdk-core`. We did a little bit of digging and it looks like v1.50.0 should include [support for `message_attributes`](https://github.com/aws/aws-sdk-ruby/blob/v1.50.0/lib/aws/sqs/queue.rb#L191-L193).

This will allow you to use your original changes from d4b9920, we just ask that you include tests to verify that the `X-GitHub-Event` attribute is passed through.

Thank you for your work so far on this. We greatly appreciate your contributions :smile:.

**Comment by jwhitcraft** [comment]
@bswinnerton,

Refactored with updated tests.

File: Gemfile.lock [hunk 2]
```diff
@@ -38,15 +38,13 @@ GEM
       thread_safe (~> 0.3, >= 0.3.4)
       tzinfo (~> 1.1)
     addressable (2.4.0)
-    aws-sdk (1.66.0)
-      aws-sdk-v1 (= 1.66.0)
+    aws-sdk (1.50.0)
```
Comment (mikemcquaid) [inline Gemfile.lock]: I think you can probably leave this at `1.66.0` actually and just tweak the `gemspec` below. Thanks!

File: lib/services/sqs_queue.rb [hunk 3]
```diff
@@ -60,6 +64,24 @@ def region
...
+    !!ENV['SQS_STUB_REQUESTS']
+  end
+
+  def aws_config
+    {
+      :region            => region,
+      :logger            => stubbed_requests? ? nil : Logger.new(STDOUT),
+      :access_key_id     => access_key,
+      :secret_access_key => secret_key,
+      :stub_requests    => stubbed_requests?
```
Comment (mikemcquaid) [inline lib/services/sqs_queue.rb]: Might want to tweak the indent on this line so it's aligned and add a trailing `,` just so more lines can be modified without a diff.

File: Gemfile.lock [hunk 4]
```diff
@@ -1,10 +1,10 @@
 PATH
   remote: .
   specs:
-    github-services (1.0.0.8dc2328)
+    github-services (1.0.0.e029d10)
       activeresource (~> 4.0.0)
       addressable (~> 2.3)
-      aws-sdk (~> 1.27)
+      aws-sdk (~> 1.50.0)
```
Comment (mikemcquaid) [inline Gemfile.lock]: This bit is correct though :+1:

**Comment by mikemcquaid** [comment]
A few notes but otherwise looking great :+1:. The failing test is fixed in #1140 so can be ignored.

**Comment by jwhitcraft** [comment]
Can we get this merged?

**Comment by bswinnerton** [comment]
Hi @jwhitcraft, 

I apologize for how long this PR has taken to push through, thank you for bearing with us. The code looks good 👍, there are just a few prerequisites to complete before this can be deployed. I hope to have this deployed in the next week.

**Comment by bswinnerton** [comment]
@jwhitcraft, we removed the `Gemfile.lock` in #1142 which I assume is the reason for the conflicts in this branch. Would you mind resolving this and then I will merge this?

**Comment by jwhitcraft** [comment]
@bswinnerton, got it updated.

**Comment by bswinnerton** [comment]
Fantastic, thanks @jwhitcraft. I intend on merging and deploying this tomorrow.

**Comment by bswinnerton** [comment]
@jwhitcraft due to the [availability issues](https://status.github.com/messages) today, I wasn't able to deploy this. I'll let you know as soon as it has been deployed.

**Comment by bswinnerton** [comment]
🎉 🚢 🎉 . This should be deployed, @jwhitcraft. If you see anything out of the ordinary, please let us know.

**Pull request merged**